osu.Game/Input/Bindings/RealmKeyBinding.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Input.Bindings;
+using osu.Game.Database;
+using Realms;
+
+namespace osu.Game.Input.Bindings
+{
+    [MapTo(nameof(KeyBinding))]
+    public class RealmKeyBinding : RealmObject, IHasGuidPrimaryKey, IKeyBinding
+    {
+        [PrimaryKey]
+        public string StringGuid { get; set; }
```

Comment:
Can this be `private set;` if everything is intended to go via `ID`, or is that not supported by Realm?

Reply:
Realm requires public `get;set;`

Reply:
Would you have anything against following the docs example for this case? Essentially something like this:
```csharp
[PrimaryKey]
public ObjectId ID { get; set; } = ObjectId.GenerateNewId();
```
(ref: https://docs.realm.io/realm-legacy-migration-guide/)

Or the string version, I'm personally not fussed either way. Just cleans this up a bit.

Reply:
No idea what an `ObjectId` is, will read into it but tentatively prefer `Guid` for compatibility as if we use anything server-side it will be guid.

Reply:
Is the part you are looking for here the default initialisation? That can still be applied for the `Guid` version. The reason I didn't do that in this PR is because I'm not sure we want to invoke that when it's immediately replaced with a response from realm or not.

Probably worth investigating the overhead of the call before doing so? Or maybe as this is a database model which shouldn't be allocated in large quantities it's deemed fine to do this?

Reply:
Yeah I was referring to the default initialisation, you're doing `ID = new Guid()` in several places and it just looks kinda awkward? It's a primary key so I'd expect it to be populated by the database, not by user code.

```
|  Method |     Mean |     Error |    StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------- |---------:|----------:|----------:|------:|------:|------:|----------:|
| NewGuid | 1.192 us | 0.0111 us | 0.0104 us |     - |     - |     - |         -
```

Reply:
Nice (how does that even work with zero alloc??). Will move it inline.

Reply:
`Guid` is a struct so no allocs.